coa/pkg/apis/v1alpha2/providers/states/memorystate/memorystate.go

```diff
@@ -231,9 +231,8 @@ func (s *MemoryStateProvider) Delete(ctx context.Context, request states.DeleteR
 	sLog.Debugf("  P (Memory State): delete state %s in namespace %s, traceId: %s", request.ID, namespace, span.SpanContext().TraceID().String())
 
 	if _, ok := s.Data[namespace]; !ok {
-		err = v1alpha2.NewCOAError(nil, fmt.Sprintf("entry '%s' is not found in namespace %s", request.ID, namespace), v1alpha2.NotFound)
-		sLog.Errorf("  P (Memory State): failed to delete %s: %+v, traceId: %s", request.ID, err, span.SpanContext().TraceID().String())
-		return err
+		sLog.Debugf("  P (Memory State): entry '%s' is not found in namespace %s, traceId: %s", request.ID, namespace, span.SpanContext().TraceID().String())
+		return nil
```

Comment:
I am OK to return 204 when deleting an object that doesn't exist but we need to have different response between successful deletion and deleting non-existent objects. Looks like the response for the above two cases are just the same.

Reply:
This is good to know in Vendor but provider only returns errors or nil in Delete action. If we would like to handle different http status, we can only do Get + Delete in Vendors. If Get returns null, return `NoContent` (needs to be added in http status const)

Reply:
The general behavior is returning 204 if deleting on non-existent objects and 200 if deletion is successful.